JavaFXUserInterface/src/main/java/SystemsController.java

```diff
@@ -134,15 +139,54 @@ public void handle(MouseEvent mouseEvent) {
             }
         });
 
+        //Adding items to the choiceBox (drop down list)
+        sortSystem.getItems().add("Default");
+        sortSystem.getItems().add("Ascending RUL");
+        sortSystem.getItems().add("Descending RUL");
+        //Default Value
+        sortSystem.setValue("Default");
+        //Listener on the sort ChoiceBox. Depending on the sort selected, all systems panes are cleared and generated again
+        //with the appropriate sort applied.
+        sortSystem.getSelectionModel().selectedItemProperty().addListener(new ChangeListener<String>() {
+            @Override
+            public void changed(ObservableValue<? extends String> observableValue, String oldValue, String newValue) {
+                switch (newValue) {
+                    case "Ascending RUL":
+                        if (thumbnailTab.isSelected()) {
+                            systemsThumbPane.getChildren().clear();
+                            generateThumbnails("Ascending RUL");
+                        }
+                        break;
+                    case "Descending RUL":
+                        if (thumbnailTab.isSelected()) {
+                            systemsThumbPane.getChildren().clear();
+                            generateThumbnails("Descending RUL");
+                        }
+                        break;
+                    default:
+                        if (thumbnailTab.isSelected()) {
+                            systemsThumbPane.getChildren().clear();
+                            generateThumbnails("Default");
+                        }
+                        break;
+                }
+            }
+        });
+
     }
 
     /**
      * Creates elements that are in the scene so the data can be displayed.
      *
      * @author Jeff
      */
-    public void generateThumbnails() {
-        for (Asset system: systems) {
+    public void generateThumbnails(String sortSelected) {
+        ObservableList<Pane> boxes = FXCollections.observableArrayList();
```

Comment:
where do you use this variable? I don't see it

Reply:
it's used at the end of generateThumbnails